server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -62,12 +62,14 @@ public LeafScoreFunction getLeafScoreFunction(LeafReaderContext ctx) throws IOEx
         leafScript._setIndexName(indexName);
         leafScript._setShard(shardId);
         return new LeafScoreFunction() {
+            ScoreScript.ExplanationHolder explanation = new ScoreScript.ExplanationHolder();
```

Comment:
We probably only want to create a new `ExplanationHolder` if explain is true for this query.

Reply:
makes sense to me; I'll poke around to see if I can figure out how to get that param down to that level

Reply:
So one problem with this is that if the plugin attempts to set an explanation even if explain isn't set to true then we will still throw an NPE.  So it may be problematic to conditionally create an ExplanationHolder.  I could create a dummy one at a higher level and pass that in, which would just be a no-op.  Thoughts on this are welcome.

Reply:
pushed up a change for this; would love some feedback on whether what I've done to both pass down that explain param and provide a dummy placeholder are reasonable.

Reply:
I believe with the refactoring I've just done this is no longer a concern; let me know if you disagree for any reason @jdconrad

Reply:
You're correct that it requires a null check in the script. I should have added a bit more context. In our [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-script-score-query.html#score-explanation), our example has a null check on the explanation holder for this reason. It's not ideal to have the user do this, but it seemed like the better trade off at the time rather than creating possibly millions of additional objects for a script that won't use them.

Reply:
Yes, that's my miss.  The docs mentioned this as did Ben in the issue raised and I missed that on my initial pass at this.